model/anomaly_nn.ipynb

```diff
@@ -1,991 +1,1322 @@
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "train data shape\n",
+            "(315602, 16, 46)\n",
+            "(315602,)\n",
+            "val data shape\n",
+            "(78901, 16, 46)\n",
+            "(78901,)\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "sPraFUqn-yUx"
+      },
+      "source": [
+        "class logDataset(Dataset):\n",
+        "    \"\"\"Log Anomaly Features Dataset\"\"\"\n",
+        "    \n",
+        "    def __init__(self, data_vec, labels=None):\n",
+        "        self.X = data_vec\n",
+        "        self.y = labels\n",
+        "        \n",
+        "    def __len__(self):\n",
+        "        return len(self.X)\n",
+        "        \n",
+        "    def __getitem__(self, idx):\n",
+        "        data_matrix = self.X[idx]\n",
+        "        \n",
+        "        if not self.y is None:\n",
+        "            return(data_matrix, self.y[idx])\n",
+        "        else:\n",
+        "            return data_matrix\n"
+      ],
+      "execution_count": 9,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Ol2-sxy5-yUy"
+      },
+      "source": [
+        "# Convert to tensort and pad matrixes to have dims of 20x48 instead of 19x48\n",
+        "# Also need to add a dimension for conv2D\n",
+        "train_data = torch.tensor(train_data, dtype=torch.float32)\n",
+        "train_data = F.pad(input=train_data, pad=(0, 0, 0, 1), mode='constant', value=0) # pad the bottom with 0s\n",
+        "train_data = np.expand_dims(train_data, axis=1)\n",
+        "test_data = torch.tensor(test_data, dtype=torch.float32)\n",
+        "test_data = F.pad(input=test_data, pad=(0, 0, 0, 1), mode='constant', value=0)\n",
+        "test_data = np.expand_dims(test_data, axis=1)\n",
+        "val_data = torch.tensor(val_data, dtype=torch.float32)\n",
+        "val_data = F.pad(input=val_data, pad=(0, 0, 0, 1), mode='constant', value=0)\n",
+        "val_data = np.expand_dims(val_data, axis=1)"
+      ],
+      "execution_count": 10,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "ShW0RdNY-yU0"
+      },
+      "source": [
+        "# Hyperparameters\n",
+        "RANDOM_SEED = 1\n",
+        "# THESE ARE ALL STARTING POINTS BASED ON LAB3\n",
+        "LEARNING_RATE = 0.001\n",
+        "BATCH_SIZE = 128\n",
+        "NUM_EPOCHS = 10\n",
+        "\n",
+        "# Architecture\n",
+        "# NUM_FEATURES = data_dims[0] * data_dims[1] # not sure this is needed currently\n",
+        "\n",
+        "# not sure we should be using an architecture with 2 classes with softmax\n",
+        "# doesn't it make more sense to have a single output node\n",
+        "# previous labs/lectures have 2 classes with softmax which is somewhat confusing\n",
```

Comment:
remove unneeded comments above - only leave comments that explain what was done as necessary